lib/Core/MemoryManager.cpp

```diff
@@ -191,8 +191,20 @@ size_t MemoryManager::getUsedDeterministicSize() {
   return nextFreeSlot - deterministicSpace;
 }
 
+ref<Expr> MemoryManager::getCacheAlignmentExpr(Expr::Width width) const {
+  return ConstantExpr::create(getCacheAlignment(), width);
+}
+
 uint64_t MemoryManager::alignToCache(uint64_t addr) const {
-  return addr - (addr % cacheAlignment);
+  return addr & ~(getCacheAlignment() - 1);
+}
+
+ref<Expr> MemoryManager::alignToCache(ref<Expr> addr) const {
```

Comment:
Why do we need to align stuff to cache lines before hitting a `PersistentState`?